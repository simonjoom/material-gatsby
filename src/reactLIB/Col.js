import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import constants from './constants';

class Col extends Component {
  render() {
    const {
      children,
      className,
      node: C = 'div',
      s,
      m,
      l,
      xl,
      offset,
      ...other
    } = this.props;

    let sizes = { s, m, l, xl };
    let classes = {
      col: true
    };

    constants.SIZES.forEach(
      size => (classes[size + sizes[size]] = sizes[size])
    );

    if (offset) {
      offset.split(' ').forEach(off => (classes['offset-' + off] = true));
    }

    return (
      <C {...other} className={cx(classes, className)}>
        {children}
      </C>
    );
  }
}

Col.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  /**
   * Columns for extra large size screens (Large Desktop Devices > 1200px)
   */
  xl: PropTypes.number,
  /**
   * Columns for large size screens (Desktop Devices > 992px)
   */
  l: PropTypes.number,
  /**
   * Columns for middle size screens (Tablet Devices > 600px)
   */
  m: PropTypes.number,
  /**
   * Columns for small size screens (Mobile Devices <= 600px)
   */
  s: PropTypes.number,
  /**
   * The node to be used for the column
   * @default div
   */
  node: PropTypes.node,
  /**
   * To offset, simply add s2 to the class where s signifies the screen
   * class-prefix (s = small, m = medium, l = large) and the number after
   * is the number of columns you want to offset by.
   */
  offset: PropTypes.string
};

export default Col;
