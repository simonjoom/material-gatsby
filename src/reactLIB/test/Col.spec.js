import React from 'react';
import { shallow } from 'enzyme';
import Col from '../src/Col';

let wrapper = shallow(<Col />);

describe('<Col />', () => {
  test('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts sizes as props', () => {
    wrapper = shallow(<Col s={4} m={6} l={8} />);
    expect(wrapper.find('.col').hasClass('s4 m6 l8')).toEqual(true);
  });
});
