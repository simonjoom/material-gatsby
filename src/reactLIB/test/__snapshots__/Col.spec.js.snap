// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> renders 1`] = `
<div
  className="col"
/>
`;
